### PR TITLE
clickhouse array support

### PR DESCRIPTION
**Pull request by gane5hvarma**
This change is [Reviewable](https://reviewable.io/reviews/rudderlabs/rudder-server/898)

### CODE REVIEW

**Review by sivashanmukh — APPROVED** [review]
LGTM

File: warehouse/clickhouse/clickhouse.go [hunk 1]
```diff
@@ -279,6 +289,52 @@ func generateArgumentString(arg string, length int) string {
...
+			if _, ok := value.(string); ok {
+				dataString = append(dataString, value.(string))
+			} else {
+				bytes, _ := json.Marshal(value)
+				dataString = append(dataString, string(bytes))
+			}
+		}
+		return dataString
+	case "array(datetime)":
+		dataTime := make([]time.Time, 0)
```
Comment (psrikanth88) [inline warehouse/clickhouse/clickhouse.go:316]: should we do like https://github.com/rudderlabs/rudder-transformer/blob/master/warehouse/index.js#L137 to keep it consistent

File: warehouse/clickhouse/clickhouse.go [hunk 2]
```diff
@@ -279,6 +289,52 @@ func generateArgumentString(arg string, length int) string {
 	return strings.Join(args, ",")
 }
 
+func castStringToArray(data string, dataType string) interface{} {
+	switch dataType {
+	case "array(int)":
+		dataInt := make([]int64, 0)
+		json.Unmarshal([]byte(data), &dataInt)
```
Comment (psrikanth88) [inline warehouse/clickhouse/clickhouse.go:296]: catch any unmarshal errors and log it?

**Review by psrikanth88 — APPROVED** [review]

File: warehouse/clickhouse/clickhouse.go (hunk 1 above)
Comment (gane5hvarma) [reply warehouse/clickhouse/clickhouse.go:316]: Internally clickhouse will convert the time value into the format set in the clickhouse settings when saving the time value into table

**Review by sivashanmukh — APPROVED** [review]

**Pull request merged**